### PR TITLE
Operators Refactor

### PR DESCRIPTION
**Pull request by abjjabjj**
- Use `options` to pass in execution options for operator calls
- @MichaelDeng03 please look at the changes I made to `demo/server`. Apparently some mypy errors made it through the cracks for `Container` that I filled just to pass linter.
- Rename `celery` to `run-async` because underlying technology not as important as what it's for.
- Adapted `orchestrator.py` to use new Operator refactor, where `run-async` is passed through to use in Operator.

### CODE REVIEW

**Comment by kgang** [comment]
Póge

**Comment by kgang** [comment]
I think we excise webapp/demo

**Comment by abjjabjj** [comment]
:open_mouth:

**Comment by abjjabjj** [comment]
ok

**Comment by abjjabjj** [comment]
i'm down lol

**Comment by abjjabjj** [comment]
let's do it in another PR

**Comment by kgang** [comment]
We can give swole lad the lines removed

File: concrete/operators.py [hunk 1]
```diff
@@ -30,7 +31,7 @@ class Developer(Operator):
         Leverage your technical expertise to create robust, scalable, and innovative AI agent orchestration systems. Apply clarity, completeness, specificity, adaptability, and creativity to deliver high-quality, impactful solutions.
     """  # noqa E501
 
-    def ask_question(self, context: str, *args, **kwargs) -> str:
+    def ask_question(self, context: str, options: dict[str, Any] = {}) -> str:
```
Comment (MichaelDeng03) [inline concrete/operators.py:34]: Why don't we just type options as `db.orm.models.OperatorOptions`? 
It seems like we're just gonna unpack and create an OperatorOptions object anyways in abstract.py. 
We can make sure option data is valid and get type checking.

File: concrete/abstract.py [hunk 2]
```diff
@@ -233,20 +259,28 @@ def __getattribute__(self, name: str) -> Any:
...
 
-        # Only wrap with qna if the result is a string.
-        # Impossible to figure out return type without running it
         def wrapped_func(*args, **kwargs):
-            result = attr(*args, **kwargs)
-            if isinstance(result, str):
-                str_func = self.qna(attr)
-                str_func.delay = partial(str_func._delay, self)
-                return str_func(*args, **kwargs)
+            options = OperatorOptions(**(self._options | kwargs.pop("options", {})))
```
Comment (MichaelDeng03) [inline concrete/abstract.py:263]: Neat! 
I wonder if we can whip up our own OperatorOptions union operator.

**Review by MichaelDeng03 — APPROVED** [review]

**Pull request merged**